processing/src/main/java/org/apache/druid/segment/join/lookup/LookupJoinMatcher.java

```diff
@@ -72,8 +73,12 @@ public ValueType defaultType()
             if (row.size() == 1) {
               return selector.lookupName(row.get(0));
             } else {
-              // Multi-valued rows are not handled by the join system right now; treat them as nulls.
-              return null;
+              // Multi-valued rows are not handled by the join system right now
+              // TODO: Remove when https://github.com/apache/druid/issues/9924 is done
+              throw new QueryException("Unsupported query",
```

Comment:
super nit: Do you think it's worth adding an `UnsupportedQueryException` class that other parts of the query stack can re-use for query operations that are unsupported? I don't know of any of these places of the top of my head, so feel free to ignore this comment. I think this is ok for now.

Reply:
I think it'll be useful. Added